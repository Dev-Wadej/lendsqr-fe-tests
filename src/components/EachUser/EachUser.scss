.eachuser {
    display: grid;
    gap: 1rem;
    grid-template-columns: repeat(6, minmax(9rem, 1fr));
    // place-content: center;
    color: rgba(84, 95, 125, 1);
    font-size: .88rem;
    padding: .88rem;
    border-radius: .2rem;
    font-weight: 400;
    @media only screen and (max-width: 1400px) {
        & {
            grid-template-columns: repeat(6, minmax(8rem, 1fr));
            font-size: .55rem;
            gap: .7rem;
        }
    }
    @media only screen and (max-width: 900px) {
        & {
            grid-template-columns: repeat(6, minmax(7rem, 1fr));
        }
    }
    li {
        a {
            color: rgba(84, 95, 125, 1);
        }
        word-wrap: break-word;
        span {
            display: inline-block;
        }
        &> :nth-child(2) {
            vertical-align: middle;
            margin-left: .5rem;
        }
        button {
            border: none;
            border-radius: 3rem;
            background: #F0F8F2;
            padding: .5rem .9rem;
            color: rgba(57, 205, 98, 1);
            @media only screen and (max-width: 1000px) {
                & {
                    font-size: 80%;
                    padding: .3rem .7rem;
                }
            }
        }
    }
    li {
        &:last-child {
            display: flex;
            justify-content: space-between;
            align-items: center;
            @media only screen and (max-width: 1200px) {
                & {
                    gap: 1rem;
                    justify-content: space-between;
                }
            }
        }
    }
}

.status-toggler-parent {
    position: relative;
}

.status-dropdown {
    position: absolute;
    z-index: 20;
    top: 0;
    right: 0;
}