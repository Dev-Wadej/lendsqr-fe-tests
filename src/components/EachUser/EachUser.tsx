import React,{useState} from 'react'
import {Link} from 'react-router-dom'
import './EachUser.scss';
import {ReactComponent as StatusToggler} from '../../data/svgs/statustoggler.svg'
import StatusDropDown from '../../components/StatusDropDown/StatusDropDown'




interface IUsersData{
  data:{
   createdAt:string;
    orgName:string
   email:string;
   id:string;
   userName:string;
  phoneNumber:string;
  }
}
const status=[
  {title:'Active',color:'#39CD62',backgroundColor:'#F3FCF6'},
{title:'Blacklisted',color:'#E4033B',backgroundColor:'#FCE6EB'},
{title:'Pending',color:'#E9B200',backgroundColor:'#FDF7E5'},
{title:'Inactive',color:'#545F7D',backgroundColor:'#F5F5F7'}
]
const randomStatusValue=()=>{
const value= Math.floor(Math.random() * status.length)
const {backgroundColor,color,title}= status[value]
return {backgroundColor,color,title}
}


const EachUser:React.FC<IUsersData> = ({data}) => {

  const [showStatusDropDown,setShowStatusDropDown]=useState(false)
  const{backgroundColor,color,title} =randomStatusValue()
  const handleStatusToggler=()=>{
    setShowStatusDropDown(prev=>!prev)
  }

  return (

    <ul className='eachuser'>
        <li>
        
          {data.orgName}
        
          </li>
        <li>
          
          {data.userName}
        
        </li>
        <li>
        
        {data.email}
        </li>
        <li>
        
        {data.phoneNumber}
        </li>
        <li>
        
        {data.createdAt}
        </li>

        <li className='status-toggler-parent' onClick={handleStatusToggler}>
            <button style={{color:color,background:backgroundColor}}>{title}</button>
            <StatusToggler/>
              {showStatusDropDown &&
              
            <StatusDropDown id={data.id}/>
              }


        </li>
    </ul>

  )
}

export default EachUser
