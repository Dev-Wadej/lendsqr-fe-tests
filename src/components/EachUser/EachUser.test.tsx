import EachUser from './EachUser'
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react';



afterEach(cleanup)
test('Each user component snapshot HTML', () => {
    const mockData={
        createdAt:'June 2022',
        orgName:'Lendsqr',
        email:'lendsr@gmail.com',
        id:'2',
        userName:'Grace',
        phoneNumber:'8199234',
    }
    const {container}= render(<EachUser data={mockData}/>)
    // console.log(container.firstChild)
  const output = renderer.create(<EachUser data={mockData}/>).toJSON();
  expect(output).toMatchSnapshot();
});