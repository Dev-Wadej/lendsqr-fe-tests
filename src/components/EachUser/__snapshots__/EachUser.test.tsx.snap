// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Each user component snapshot HTML 1`] = `
<ul
  className="eachuser"
>
  <li>
    Lendsqr
  </li>
  <li>
    Grace
  </li>
  <li>
    lendsr@gmail.com
  </li>
  <li>
    8199234
  </li>
  <li>
    June 2022
  </li>
  <li
    className="status-toggler-parent"
    onClick={[Function]}
  >
    <button
      style={
        Object {
          "background": "#F3FCF6",
          "color": "#39CD62",
        }
      }
    >
      Active
    </button>
    <svg>
      statustoggler.svg
    </svg>
  </li>
</ul>
`;
